platforms/software/dependency-management/src/main/java/org/gradle/internal/component/ExternalConfigurationNotFoundException.java

```diff
@@ -14,12 +14,18 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.component.model;
+package org.gradle.internal.component;
 
 import org.apache.commons.lang.StringUtils;
 import org.gradle.api.artifacts.component.ComponentIdentifier;
 
-public class ExternalConfigurationNotFoundException extends RuntimeException {
+/**
+ * This exception is used to indicate that a named configuration for an <strong>Ivy</strong> or <strong>Maven</strong>
+ * dependency wasn't found when resolving a Graph.
+ *
+ * For Maven, the configuration is always the single variant that uses the name "default".
```

Comment:
🤔 I would check the code again. I do not think this is always true, it is possible that we also check for `compile` and `runtime`. 

In any case, I question whether we should be documenting this here. I think we should try to use class-level javadoc to describe the functionality/behavior class in question, and not how it is used. I think its okay to document how a class _should_ be used, but if we document how it is currently used, that makes it very easy for logic and doc to get out of sync

Reply:
That's fair.  This exception is always thrown when _requesting_ "default", but that's not exactly what my javadoc says and you're right, it shouldn't matter anyways.  I'll remove this.